src/torque-offset-estimator.cpp

```diff
@@ -0,0 +1,265 @@
+        addCommand("getSensorOffsets",
+                   makeDirectGetter(*this,&jointTorqueOffsets,
+                                    docDirectGetter("Return the computed sensor offsets",
+                                                    "vector")));
+
+        encSignals.clear();
+        accSignals.clear();
+        gyrSignals.clear();
+        tauSignals.clear();
+        
+      }
+
+
+      /* --- COMMANDS ---------------------------------------------------------- */
+      /* --- COMMANDS ---------------------------------------------------------- */
+      /* --- COMMANDS ---------------------------------------------------------- */
+      void TorqueOffsetEstimator::init(const std::string &urdfFile,
+                                       const Eigen::Matrix4d& _m_torso_X_imu)
+      {
+        try {
+          se3::urdf::buildModel(urdfFile,se3::JointModelFreeFlyer(),m_model);
+          assert(m_model.nq == N_JOINTS+7);
+          assert(m_model.nv == N_JOINTS+6);
+          jointTorqueOffsets.resize(N_JOINTS);
+          jointTorqueOffsets.setZero();
+        }
+        catch (const std::exception& e) 
+        { 
+          std::cout << e.what();
+          return SEND_MSG("Init failed: Could load URDF :" + urdfFile, MSG_TYPE_ERROR);
+        }
+        m_data = new se3::Data(m_model);
+        m_torso_X_imu.rotation() = _m_torso_X_imu.block<3,3>(0,0);
+        m_torso_X_imu.translation() = _m_torso_X_imu.block<3,1>(0,3);
+      }
+
+      void TorqueOffsetEstimator::computeOffset(const int& nIterations, const double& _epsilon)
+      {
+        if (sensor_offset_status == PRECOMPUTATION) {
+          SEND_MSG("Starting offset computation with no. iterations:"+ nIterations, MSG_TYPE_DEBUG);
+          n_iterations = nIterations;
+          sensor_offset_status = INPROGRESS;
+        }
+        else if (sensor_offset_status == INPROGRESS) {
+          SEND_MSG("Collecting input signals. Please keep the graph running", MSG_TYPE_WARNING_STREAM);
+        }
+        else if (sensor_offset_status == COMPUTED) {
+          SEND_MSG("Recomputing offset with no. iterations:"+ nIterations, MSG_TYPE_DEBUG);
+
+          // resetSavedSignals();
+          encSignals.clear();
+          accSignals.clear();
+          gyrSignals.clear();
+          tauSignals.clear();
+
+          jointTorqueOffsets.setZero();
+
+          n_iterations = nIterations;
+          epsilon = _epsilon;
+          sensor_offset_status = INPROGRESS;
+        }
+        else { //should not happend
+          SEND_MSG("Problem in computeOffset sensor_offset_status", MSG_TYPE_ERROR);
+          assert(false && "Problem in computeOffset sensor_offset_status");
+        }
+        return;
+      }
+      
+      DEFINE_SIGNAL_INNER_FUNCTION(collectSensorData, dummy)
+      {
+        if (sensor_offset_status == PRECOMPUTATION)  {
+          SEND_MSG("Currently Precomputation. Not collecting signals", MSG_TYPE_DEBUG);
```

Comment:
The STREAM msg types are supposed to be called at each control cycle (but then they are not printed at each control cycle). The other msg types are instead printed every time, so you shouldn't use them inside the signal computations. 

In this specific case I'd say you should just remove this print.